src/main/scala/com/fulcrumgenomics/bam/ClipBam.scala

```diff
@@ -58,14 +56,26 @@ import scala.collection.immutable.IndexedSeq
     |pair information can be reset across all reads for the template.  Post-clipping the reads are
     |resorted into coordinate order, any existing `NM`, `UQ` and `MD` tags are repaired, and the output is
     |written in coordinate order.
+    |
+    |Three clipping modes are supported:
+    |1. `Soft` - soft-clip the bases and qualities.
+    |2. `SoftWithMask` - soft-clip and mask the bases and qualities (make bases Ns and qualities the minimum).
+    |3. `Hard` - hard-clip the bases and qualities.
+    |
+    |The `--upgrade-clipping` parameter will convert all existing clipping in the input to given more stringent mode:
+    |from `Soft` to either `SoftWithMask` or `Hard`, and `SoftWithMask` to `Hard`. In all other cases, clipping remains
+    |the same prior to applying any other clipping criteria.
   """)
 class ClipBam
 ( @arg(flag='i', doc="Input SAM or BAM file of aligned reads in coordinate order.") val input: PathToBam,
   @arg(flag='o', doc="Output SAM or BAM file.") val output: PathToBam,
   @arg(flag='m', doc="Optional output of clipping metrics.") val metrics: Option[FilePath] = None,
   @arg(flag='r', doc="Reference sequence fasta file.") val ref: PathToFasta,
-  @arg(flag='s', doc="Soft clip reads instead of hard clipping.") val softClip: Boolean = false,
+  @deprecated("Use clipping-mode instead.", since="0.4.0")
+  @arg(flag='s', doc="Soft clip reads instead of hard clipping.", mutex=Array("clippingMode")) val softClip: Boolean = false,
```

Comment:
I'm sorta surprised this mutex works when softClip has a default value, but I checked it out and built and it does!

Reply:
I had to build it too so that I could verify.  It makes sense to me at least, mutex doesn't ask if there _are_ values, only that the user shouldn't specify both.